From 80ddbc5081a66ab9d3d494044ddbe3a7c94efbed Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Tue, 21 Oct 2014 17:00:08 +0100
Subject: [PATCH 054/162] staging: etnaviv: remove shifting and bitwise or-ing
 of GPU addresses

Remove support for shifting, or bitwise or-ing of constants with GPU
addresses.  The Vivante GPU does not need this support, so we can
remove these additional operations.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_gem_submit.c | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_gem_submit.c b/drivers/staging/etnaviv/etnaviv_gem_submit.c
index c32fb44..cdc0bb5 100644
--- a/drivers/staging/etnaviv/etnaviv_gem_submit.c
+++ b/drivers/staging/etnaviv/etnaviv_gem_submit.c
@@ -259,6 +259,13 @@ static int submit_reloc(struct etnaviv_gem_submit *submit, struct etnaviv_gem_ob
 			return -EINVAL;
 		}
 
+		/* Vivante hardware has no need for shifts or bitwise or-ing */
+		if (submit_reloc.shift || submit_reloc.or) {
+			DRM_ERROR("invalid shift or bitwise or at reloc %u\n",
+				  i);
+			return -EINVAL;
+		}
+
 		ret = submit_bo(submit, submit_reloc.reloc_idx, &bobj, &iova);
 		if (ret)
 			return ret;
@@ -269,14 +276,7 @@ static int submit_reloc(struct etnaviv_gem_submit *submit, struct etnaviv_gem_ob
 			return -EINVAL;
 		}
 
-		iova += submit_reloc.reloc_offset;
-
-		if (submit_reloc.shift < 0)
-			iova >>= -submit_reloc.shift;
-		else
-			iova <<= submit_reloc.shift;
-
-		ptr[off] = iova | submit_reloc.or;
+		ptr[off] = iova + submit_reloc.reloc_offset;
 
 		last_offset = off;
 	}
-- 
2.6.2

